app/services/travel_to_work_area_and_london_borough_service.rb

```diff
@@ -1,20 +1,60 @@
 class TravelToWorkAreaAndLondonBoroughService
-  def self.add_travel_to_work_area_and_london_borough(site:)
-    url = URI("#{Settings.mapit_url}/point/4326/#{site.longitude},#{site.latitude}?type=TTW,LBO&api_key=#{Settings.mapit_api_key}")
-    response = Net::HTTP.get(url)
-    json = JSON.parse(response)
-
-    ttw_area = json.dig(json.keys.first, "name")
-
-    london_borough = if ttw_area == "London"
-                       json
-                       .dig(json.keys.second, "name")
-                       .gsub(/ Borough Council| City Council| Corporation/, "")
-                     end
-
-    site.update_column("travel_to_work_area", ttw_area)
-    site.update_column("london_borough", london_borough)
-  rescue StandardError
-    false
+  include ServicePattern
+
+  def initialize(site:)
+    @site = site
+  end
+
+  def call
+    ttw_area = get(:travel_to_work_area)
+
+    if ttw_area == "London"
+      london_borough = get(:london_borough)
+
+      if london_borough
+        site.travel_to_work_area = "London"
+        site.london_borough = london_borough
+        site.save!(validate: false)
+      end
+    elsif ttw_area
+      site.travel_to_work_area = ttw_area
+      site.london_borough = nil
+      site.save!(validate: false)
+    else
+      false
+    end
+  end
+
+private
+
+  attr_reader :site
+
+  def get(attribute)
+    url = if attribute == :travel_to_work_area
+            travel_to_work_url
+          else
+            london_borough_url
+          end
+
+    response = Net::HTTP.get_response(url)
+
+    if response.code == "200"
```

Comment:
If its not a `200` do we need to log it?

ie.
Out of quotas?
Out of service?
Out of cash?

Reply:
Yeah good point.

I'm now generating an error that is captured by Sentry when the Mapit API returns a 403 or 5XX status

https://github.com/DFE-Digital/teacher-training-api/pull/1786/commits/963aa6fe0fcace559a69c3de4466d4a4902a8f44